libs/seiscomp/datamodel/utils.cpp

```diff
@@ -379,18 +379,18 @@ bool by_Z_desc_and_code_asc(const ComponentAxis &axis1, const ComponentAxis &axi
 }
 
 
-bool getThreeComponents(ThreeComponents &res, const SensorLocation *loc, const char *streamCode, const Core::Time &time) {
+bool getThreeComponents(ThreeComponents &tc, const SensorLocation *loc, const char *streamCode, const Core::Time &time) {
 	int len = strlen(streamCode);
 
-	res = ThreeComponents();
-
-	std::vector<ComponentAxis> comps;
+	tc = ThreeComponents();
 
+	// First find matching streams based on code and time
+	std::vector<Stream*> matchingStreams;
 	for ( size_t i = 0; i < loc->streamCount(); ++i ) {
 		Stream *stream = loc->stream(i);
 
 		try {
-			if ( stream->end() <= time ) {
+			if ( stream->end() < time ) {
```

Comment:
I don't think that this is correct. End time is not inclusive: start <= time < end.

Reply:
I agree with you.

Reply:
I agree with you.